src/api/users/doctor/doctor.controller.ts

```diff
@@ -155,11 +159,37 @@ export class DoctorController extends BaseUserController {
     deleteByUserId = async (request: express.Request, response: express.Response): Promise<void> => {
         try {
             const userId: string = await DoctorValidator.delete(request);
+            const currentUserId = request.currentUser.UserId;
+            const doctor = await this._service.getByUserId(userId);
+            const personId = doctor.User.PersonId;
+            if (!doctor) {
+                throw new ApiError(404, 'Doctor account does not exist!');
+            }
+            if (currentUserId !== userId) {
+                throw new ApiError(403, 'You do not have permissions to delete this doctor account.');
+            }
             const user = await this._userService.getById(userId);
             if (user == null) {
                 throw new ApiError(404, 'User not found.');
             }
             await this.authorizeOne(request, userId, user.TenantId);
+            let deleted = await this._cohortService.removeUserFromAllCohorts(userId);
+            deleted = await this._userService.delete(userId);
```

Comment:
You are deleting 'User' before you are deleting 'Doctor'...It should be other way around.
- Same for the patient...
The thumb rule is
- While deleting the sequence of deletion should be reverse of that of the creation sequence.